### PR TITLE
Fix broken pip due to cryptography/setuptools/pip mismatch

### LINKED ISSUE #1081: Installation broken on brand-new Digital Ocean box
This happens to be every time:

1. Set up a fresh Digital Ocean box with 14.04.
2. Do apt-get update && apt-get upgrade.
3. Run curl -s https://mailinabox.email/setup.sh | sudo bash

It fails here:

```
FAILED: pip3 install --upgrade rtyaml email_validator>=1.0.0 free_tls_certificates>=0.1.3 idna>=2.0.0 cryptography>=1.0.2 boto psutil
.....
.....
ImportError: No module named 'packaging'
```
Full error here: https://gist.github.com/miracle2k/3f84f408db3a3143a01ad27c2645554c

Afterwards, pip3 is completely broken:

```
root@mail1:~# pip3
Traceback (most recent call last):
  File "/usr/bin/pip3", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/usr/local/lib/python3.4/dist-packages/pkg_resources/__init__.py", line 70, in <module>
    import packaging.version
ImportError: No module named 'packaging'
```

### PR DESCRIPTION
**Pull request by cromulus**
fixes #1081 and #1080

I was able to replicate and adding `setuptools` to the initial pip3 installation seemed to fix it.

### CODE REVIEW

**Comment by cromulus** [comment]
Yeah, this is much more of a problem than simply installing setuptools in advance. I've filed a bug with python cryptography and cffi, as it appears this is an upstream issue.

**Comment by cromulus** [comment]
Looks like it's an upstream issue with pip: https://github.com/pypa/pip/issues/4253

**Comment by cromulus** [comment]
I think this works. At least for me, on brand new Digital Ocean ubuntu 14.04.5 machine.

**Comment by axd** [comment]
You should add `python3-requests`. I found some trouble without this package in the past.

**Comment by cromulus** [comment]
@JoshData @yodax: Would you mind taking a peek at this PR?

As far as I can tell, the issue this PR addresses is breaking many new installations.

**Comment by JoshData** [comment]
This looks like it was a real upstream mess. Thanks for doing the diagnosing. I'll take a look this weekend.

**Comment by cromulus** [comment]
Yeah, as far as I can tell, cryptography changed their setuptools requirements which broke all the things. 

Thanks for digging in!

**Comment by yodax** [comment]
I haven't had a chance to look into this. Hopefully I will have some time tomorrow.

**Comment by cromulus** [comment]
I'm doing a bit more work on it. I think I have a solution to both new installations AND fixing people that upgraded in the midst of this issue.

File: setup/management.sh [hunk 1]
```diff
@@ -15,7 +15,19 @@ apt_install build-essential libssl-dev libffi-dev python3-dev
...
-hide_output pip3 install --upgrade setuptools
+
+# the below fixes the issue with an older version of pip3 and setup tools
+# breaking pip3 via a race condition
+# see https://github.com/pypa/pip/issues/4253
+# and
+# https://github.com/mail-in-a-box/mailinabox/issues/1086
+
+# need to use hide output, because we have to re-install this.
+hide_output apt-get install -y --reinstall python-pkg-resources
```
Comment (yodax) [inline setup/management.sh]: Do we need to do this every time or is there a way to detect if it's necessary?

Comment (cromulus) [inline setup/management.sh]: Hard to say. 

Each installation may have a different version of setuptools, depending on when they last installed the cryptography library. 

My guess is that the right way to do this is to lock libraries at specific versions and use virtualenv or something similar. 

I'm not tremendously familiar with the python ecosystem.

**Comment by yodax** [comment]
I have looked into this some more. This problem isn't unique to DO, it happens on Scaleway as well (and I assume everywhere since it's a pip issue). 

The way I found to fix a broken system and a clean install is to run: 

```
apt-get purge -y python3-pip
wget https://bootstrap.pypa.io/get-pip.py
python3 ./get-pip.py
apt-get install python3-pip
```

I'm not sure if we should incorporate a fix for this in MIAB since it is an upstream issue. Deleting and installing pip every time seems bad. Also I can't find a hash on the site to verify against (t.b.h. I haven't looked very hard 😋)

**Comment by JoshData** [comment]
Thanks guys. This all was very helpful. I don't think this is the right approach (looks like you're installing packages for the wrong version of python) and am working on it now.

**Pull request closed without merging**